[Bug]vela-core will crash when rolloutBatches is empty

**Describe the bug**
Now, AppRollout's `spec.rolloutPlan.rolloutBatches` is optional but if we create a AppRollout without rolloutBatches 
KubeVela controller will crash

**To Reproduce**
1. apply the appRollout yaml
```yaml
apiVersion: core.oam.dev/v1beta1
kind: AppRollout
metadata:
  name: rolling-example
spec:
  targetAppRevisionName: test-rolling-v1
  componentList:
    - metrics-provider
  rolloutPlan:
    rolloutStrategy: "IncreaseFirst"
    targetSize: 2
```
2. vela core will crash, check the log
```shell
E0510 07:22:50.819047       1 runtime.go:78] Observed a panic: runtime.boundsError{x:-1, y:0, signed:true, code:0x0} (runtime error: index out of range [-1])
goroutine 573 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1c5a220, 0xc00a4f67e0)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1c5a220, 0xc00a4f67e0)
        /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/oam-dev/kubevela/pkg/controller/common/rollout/workloads.verifyBatchesWithScale(0xc007a8c180, 0x1, 0x3, 0x0, 0x0)
        /workspace/pkg/controller/common/rollout/workloads/common.go:88 +0x4ba
github.com/oam-dev/kubevela/pkg/controller/common/rollout/workloads.(*DeploymentScaleController).VerifySpec(0xc00ae9eb60, 0x20a0060, 0xc00757b2c0, 0x0, 0x0, 0x0)
        /workspace/pkg/controller/common/rollout/workloads/deployment_scale_controller.go:91 +0x2f0
github.com/oam-dev/kubevela/pkg/controller/common/rollout.(*Controller).Reconcile(0xc0007c4050, 0x20a0060, 0xc00757b2c0, 0xc0004ff600, 0x12a05f200, 0xc00ae9e9a0)
        /workspace/pkg/controller/common/rollout/rollout_plan_controller.go:115 +0x782
github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/applicationrollout.(*Reconciler).DoReconcile(0xc005f7cfc0, 0x20a0060, 0xc00757b2c0, 0xc0004ff680, 0x0, 0xc00670f4e0, 0x0, 0x1)
        /workspace/pkg/controller/core.oam.dev/v1alpha2/applicationrollout/applicationrollout_controller.go:228 +0xc41
github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application.(*appHandler).handleRollout(0xc00092aa80, 0x20a0060, 0xc00757b2c0, 0xc00a792800, 0xc000568000, 0xc0096ed018, 0x1)
        /workspace/pkg/controller/core.oam.dev/v1alpha2/application/apply.go:748 +0x40e
github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application.(*Reconciler).Reconcile(0xc009b85ab0, 0xc00a82d6f9, 0x7, 0xc00a82d6e0, 0xc, 0xc00689cde0, 0xc009904750, 0xc0099046c8, 0xc0099046c0)
        /workspace/pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:178 +0x2355
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0009b8360, 0x1b9e220, 0xc008da1840, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:235 +0x284
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0009b8360, 0x203000)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:209 +0xae
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0009b8360)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:188 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc005fb9610)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc005fb9610, 0x2059660, 0xc009f450e0, 0x1, 0xc00045f200)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc005fb9610, 0x3b9aca00, 0x0, 0x1, 0xc00045f200)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc005fb9610, 0x3b9aca00, 0xc00045f200)
        /go/pkg/mod/k8s.io/apimachinery@v0.18.8/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:170 +0x411
panic: runtime error: index out of range [-1] [recovered]
        panic: runtime error: index out of range [-1]

```


**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

**Screenshots**
<!--
If applicable, add screenshots to help explain your problem.
-->

**Cluster information**
<!--
Describe your kubernetes cluster information.
- Kubernetes Version [e.g. 1.16.9]
-->

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/pkg/controller/common/rollout/workloads/common.go
+++ b/pkg/controller/common/rollout/workloads/common.go
@@ -29,6 +29,10 @@ import (
 // verifyBatchesWithRollout verifies that the the sum of all the batch replicas is valid given the total replica
 // each batch replica can be absolute or a percentage
 func verifyBatchesWithRollout(rolloutSpec *v1alpha1.RolloutPlan, totalReplicas int32) error {
+	// If rolloutBatches length equal to zero will cause index out of bounds panic, guarantee don't crash whole vela controller
+	if len(rolloutSpec.RolloutBatches) == 0 {
+		return fmt.Errorf("the rolloutPlan must have batches")
+	}
 	// if not set, the sum of all the batch sizes minus the last batch cannot be more than the totalReplicas
 	totalRollout := 0
 	for i := 0; i < len(rolloutSpec.RolloutBatches)-1; i++ {
@@ -57,6 +61,10 @@ func verifyBatchesWithRollout(rolloutSpec *v1alpha1.RolloutPlan, totalReplicas i
 
 // verifyBatchesWithScale verifies that executing batches finally reach the target size starting from original size
 func verifyBatchesWithScale(rolloutSpec *v1alpha1.RolloutPlan, originalSize, targetSize int) error {
+	// If rolloutBatches length equal to zero will cause index out of bounds panic, guarantee don't crash whole vela controller
+	if len(rolloutSpec.RolloutBatches) == 0 {
+		return fmt.Errorf("the rolloutPlan must have batches")
+	}
 	totalRollout := originalSize
 	for i := 0; i < len(rolloutSpec.RolloutBatches)-1; i++ {
 		rb := rolloutSpec.RolloutBatches[i]

--- a/pkg/controller/common/rollout/workloads/common_test.go
+++ b/pkg/controller/common/rollout/workloads/common_test.go
@@ -372,6 +372,11 @@ func Test_VerifyBatchesWithScaleFailCases(t *testing.T) {
 			originalSize: 16,
 			targetSize:   10,
 		},
+		"empty rollingBatches": {
+			rolloutSpec:  &v1alpha1.RolloutPlan{},
+			targetSize:   3,
+			originalSize: 2,
+		},
 	}
 	for name, tt := range tests {
 		t.Run(name, func(t *testing.T) {

--- a/pkg/controller/common/rollout/workloads/common_test.go
+++ b/pkg/controller/common/rollout/workloads/common_test.go
@@ -19,6 +19,8 @@ package workloads
 import (
 	"testing"
 
+	"k8s.io/utils/pointer"
+
 	"k8s.io/apimachinery/pkg/util/intstr"
 
 	"github.com/oam-dev/kubevela/apis/standard.oam.dev/v1alpha1"
@@ -213,6 +215,15 @@ func TestVerifyBatchesWithRolloutRelaxed(t *testing.T) {
 		t.Errorf("verifyBatchesWithRollout() = %v, want error", nil)
 	}
 	if err := verifyBatchesWithRollout(rolloutOverFlowSpec, 0); err == nil {
+		t.Errorf("verifyBatchesWithRollout() = %v, want error", nil)
+	}
+}
+
+func TestVerifyEmptyRolloutBatches(t *testing.T) {
+	plan := &v1alpha1.RolloutPlan{
+		TargetSize: pointer.Int32Ptr(2),
+	}
+	if err := verifyBatchesWithRollout(plan, 3); err == nil {
 		t.Errorf("verifyBatchesWithRollout() = %v, want error", nil)
 	}
 }

--- a/pkg/webhook/core.oam.dev/v1alpha2/application/validating_handler_test.go
+++ b/pkg/webhook/core.oam.dev/v1alpha2/application/validating_handler_test.go
@@ -128,4 +128,24 @@ var _ = Describe("Test Application Validator", func() {
 		resp := handler.Handle(ctx, req)
 		Expect(resp.Allowed).Should(BeFalse())
 	})
+
+	It("Test Application Validator rolloutPlan [error]", func() {
+		req := admission.Request{
+			AdmissionRequest: admissionv1beta1.AdmissionRequest{
+				Operation: admissionv1beta1.Create,
+				Resource:  metav1.GroupVersionResource{Group: "core.oam.dev", Version: "v1alpha2", Resource: "applications"},
+				Object: runtime.RawExtension{
+					Raw: []byte(`
+{"kind":"Application","metadata":{"name":"test-rolling","annotations":null},
+"spec":{"components":[{"name":"metrics-provider","type":"worker",
+"properties":{"cmd":["./podinfo","stress-cpu=3.0"],
+"image":"stefanprodan/podinfo:4.0.6","port":8080}}],
+"rolloutPlan":{"rolloutStrategy":"IncreaseFirst","targetSize":3}}}
+`),
+				},
+			},
+		}
+		resp := handler.Handle(ctx, req)
+		Expect(resp.Allowed).Should(BeFalse())
+	})
 })

--- a/pkg/webhook/core.oam.dev/v1alpha2/application/validation.go
+++ b/pkg/webhook/core.oam.dev/v1alpha2/application/validation.go
@@ -24,6 +24,7 @@ import (
 	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
 	"github.com/oam-dev/kubevela/pkg/appfile"
 	"github.com/oam-dev/kubevela/pkg/oam"
+	"github.com/oam-dev/kubevela/pkg/webhook/common/rollout"
 )
 
 // ValidateCreate validates the Application on creation
@@ -43,6 +44,9 @@ func (h *ValidatingHandler) ValidateCreate(ctx context.Context, app *v1beta1.App
 	}
 	if v := app.GetAnnotations()[oam.AnnotationAppRollout]; len(v) != 0 && v != "true" {
 		componentErrs = append(componentErrs, field.Invalid(field.NewPath("annotation:app.oam.dev/rollout-template"), app, "the annotation value of rollout-template must be true"))
+	}
+	if app.Spec.RolloutPlan != nil {
+		componentErrs = append(componentErrs, rollout.ValidateCreate(h.Client, app.Spec.RolloutPlan, field.NewPath("rolloutPlan"))...)
 	}
 	return componentErrs
 }

--- a/test/e2e-test/application_test.go
+++ b/test/e2e-test/application_test.go
@@ -165,4 +165,12 @@ var _ = Describe("Application Normal tests", func() {
 		newApp.Namespace = namespaceName
 		Expect(k8sClient.Create(ctx, &newApp)).ShouldNot(BeNil())
 	})
+
+	It("Test app have empty rollingBatches rolloutPlan", func() {
+		By("Apply an application")
+		var newApp v1beta1.Application
+		Expect(common.ReadYamlToObject("testdata/app/app6.yaml", &newApp)).Should(BeNil())
+		newApp.Namespace = namespaceName
+		Expect(k8sClient.Create(ctx, &newApp)).ShouldNot(BeNil())
+	})
 })

--- a/test/e2e-test/testdata/app/app6.yaml
+++ b/test/e2e-test/testdata/app/app6.yaml
@@ -1,0 +1,17 @@
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: test-rolling
+spec:
+  components:
+    - name: metrics-provider
+      type: worker
+      properties:
+        cmd:
+          - ./podinfo
+          - stress-cpu=3.0
+        image: stefanprodan/podinfo:4.0.6
+        port: 8080
+  rolloutPlan:
+    rolloutStrategy: "IncreaseFirst"
+    targetSize: 3

